controllers/gateway_controller.go

```diff
@@ -36,11 +38,25 @@ const gatewayServiceLabel = "konghq.com/owned-by-gateway"
 type GatewayReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
+	Log    logr.Logger
 }
 
 func (r *GatewayReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&gatewayv1beta1.Gateway{}).
+		WithEventFilter(predicate.NewPredicateFuncs(func(obj client.Object) bool {
```

Comment:
You should put this predicate as the variadic argument in the `For(&gatewayv1beta1.Gateway{})`, because if we set it globally here, it will be applied to all other resources that will be watched/owned by this reconcile in the future.

Reply:
Also, please define this `func(obj client.Object) bool {...}` as a function. You can take inspiration from here: https://github.com/Kong/kubernetes-ingress-controller/blob/fc162862696c8eb4c400a73336abc0017af78f23/internal/controllers/gateway/gateway_controller.go#L91